### PR TITLE
- Added DTO to the models CinemaHall, Movie, MovieSession

### PR DESCRIPTION
**Pull request by MykolaFatkullin**
- Added MovieSessionController, CinemaHallController, MovieController
- Added mappers to the DTOs

### CODE REVIEW

**Review by mfedenko09 — APPROVED** [review]

File: src/main/java/com/tickets/controller/MovieSessionController.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            @RequestParam Long id, @RequestParam @DateTimeFormat(pattern = "dd.MM.yyyy")
+                    LocalDate date) {
+        return movieSessionService.findAvailableSessions(id, date)
+                .stream()
+                .map(movieSessionMapper::mapToDto)
+                .collect(Collectors.toList());
+    }
+
+    @PutMapping
+    public void updateSession(@RequestParam Long id,
```
Comment (boroda4436) [inline src/main/java/com/tickets/controller/MovieSessionController.java]: is it ok to use `@RequestParam` according to [REST Resource Naming Guide](https://restfulapi.net/resource-naming/)?

File: src/main/java/com/tickets/controller/MovieSessionController.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    @PostMapping
+    public void addMovieSession(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession = movieSessionMapper.dtoToMap(movieSessionRequestDto);
+        movieSessionService.add(movieSession);
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAllAvailable(
+            @RequestParam Long id, @RequestParam @DateTimeFormat(pattern = "dd.MM.yyyy")
```
Comment (boroda4436) [inline src/main/java/com/tickets/controller/MovieSessionController.java]: Is this `id` an identifier of MovieSession? If yes why you are using RequestParam? If no, let's rename

File: src/main/java/com/tickets/service/dto/impl/MovieSessionMapperImpl.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package com.tickets.service.dto.impl;
```
Comment (boroda4436) [inline src/main/java/com/tickets/service/dto/impl/MovieSessionMapperImpl.java]: better to name package `mapper` or `mappers` instead of `dto`

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/com/tickets/controller/MovieSessionController.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    public List<MovieSessionResponseDto> getAllAvailable(
+            @RequestParam Long movieId, @RequestParam @DateTimeFormat(pattern = "dd.MM.yyyy")
+                    LocalDate date) {
+        return movieSessionService.findAvailableSessions(movieId, date)
+                .stream()
+                .map(movieSessionMapper::mapToDto)
+                .collect(Collectors.toList());
+    }
+
+    @PutMapping("{id}")
```
Comment (boroda4436) [inline src/main/java/com/tickets/controller/MovieSessionController.java]: good, better write /{id}

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**